blockchain/chain.go

```diff
@@ -1331,6 +1331,18 @@ func (b *BlockChain) reorganizeChain(detachNodes, attachNodes *list.List) error
 	b.sendNotification(NTReorganization, reorgData)
 	b.chainLock.Lock()
 
+	// Send a notification announcing the start of the chain reorganization.
+	b.chainLock.Unlock()
+	b.sendNotification(NTChainReorgStarted, &ChainReorgStarted{})
```

Comment:
See other comments.  No need to create an empty heap object that has to be GC'd.   Just use `nil`.

```Go
b.sendNotification(NTChainReorgStarted, nil)
```